src/lstm/functions.h

```diff
@@ -200,8 +200,9 @@ inline void SoftmaxInPlace(int n, T *inout) {
     inout[i] = prob;
   }
   if (prob_total > 0) {
+    T inv_prob_total = 1 / prob_total;
     for (int i = 0; i < n; i++) {
-      inout[i] /= prob_total;
+      inout[i] *= inv_prob_total;
```

Comment:
Isn't this kind of optimization something which a good compiler should do automatically?